Don't suggest private methods

gnoe opened the pull request:
To avoid errors on Travis with line ending.

This branch has the same fix for showing private and protected methods and properties out of the context.

Fixes #469

gnoe committed:
[PR] Intellisense autocomplete also suggests private or protected method and properties

Checking the visibility and the context from where the node is: inside a method declaration or after an instantiation

gnoe committed:
Adding netbeans config to gitignore

gnoe committed:
Resolving node ancestor in the CompletionProvider, so it's resolved outside the loop

gnoe committed:
Update .gitignore

Ignoting Netbeans conf

gnoe committed:
Minor changes

gnoe committed:
Merge branch 'master' of https://github.com/gnoe/php-language-server into fix/visibility

gnoe committed:
Merging with master

gnoe commented:
Yes, I can.
Should I add it to tests\Server\TextDocument\CompletionTest.php? 
I tested this launching the php-intellisense extension pointing to my local copy of the language server.
I will write the test.

Travis keeps failing because of the line ending.

felixfbecker commented:
What editor are you using? You should configure it to trim trailing whitespace (or delete it manually)

gnoe commented:
I use VS Code and Netbeans, I configured VS Code.
Thanks!
Next commit with the test should be fixed

gnoe committed:
Trimmed Trailing Whitespace

gnoe committed:
Visibility Test and Changing line ending from CRLF -> LF

Added CompletionWithVisibilityTest.php
- checks visibility of a property or method call (only public)
- checks visibility of a Class (can see all methods and properties)
- checks visibility of a child class (public and protected)

gnoe committed:
Fixing phpcs errors

gnoe committed:
CompletionTest fixed

gnoe committed:
Decided to move visibility files apart

Other tests were failing because I modifed the global_symbols.php file

gnoe committed:
EOL

gnoe committed:
Merge branch 'master' into fix/eol

gnoe commented:
I did merge the branch already, sorry, I have to get used to the code. I removed the part of the static checking. I would like to test it better.

gnoe committed:
[FIX] Checked changes and tested again

gnoe committed:
Merge branch 'master' into fix/eol

gnoe committed:
Merge branch 'master' into fix/eol
